### PR TITLE
Fix various JSON corner case

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
1. Fix implicit type casting behavior for binary predicate
2. Forbid using JSON type in unique/primary/aggregate key

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_clang-format

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java [hunk 1]
```diff
@@ -93,8 +93,15 @@ public static Expr addCastExpr(Expr expr, Type targetType) {
...
         Preconditions.checkState(types.size() > 0);
         Type compatibleType = types.get(0);
+        // TODO(mofei) support it
+        if (compatibleType.isJsonType()) {
+            throw new SemanticException("InPredicate of JSON is not supported");
+        }
 
         for (int i = 1; i < types.size(); i++) {
+            if (types.get(i).isJsonType()) {
+                throw new SemanticException("InPredicate of JSON is not supported");
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java:103]: Relevant semantic checking of expressions should be done in ExpressionAnalyzer, not in TypeManager

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java]: fixed

**Review by Youngwb — APPROVED** [review]

**Comment by mofeiatwork** [comment]
run starrocks_clang-format

**Review by Seaven — APPROVED** [review]

**Pull request merged**